### DISCUSSION TITLE
Successful implimentation of tqdm progress bar in pdfplumber.

### DISCUSSION THREAD

**Discussion by LeeSugden** [jsvine/pdfplumber#541] [Q&A] [answered]
Hi, has anyone successfully implimented a tqdm progress bar with the progress out extract from a .pdf, For instance i would like my progress bar to add 1% when it has done 1/100 pages. I would find this super useful as i regularly have documents with thousands of pages and struggle to see where at in the process my code is. Until it reaches the next step then i can `print("it was successful")`. Python3.

I currently use the following method:

```
for page in pdf.pages:
    print(page.extract_text())
```

However i was unsure how to impliment tqdm without a for loop which is in the tqdm documentation.

So i tried the following code however it doesnt seem to do anything progress bar wise.

```
with pdfplumber.open(path) as pdf:
    for i in tqdm(range(len(pdf.pages))):
        print(pdf.pages[i].extract_text())
```

Any help of example code of it work that i can adjust and ammend accordingly would be really helpful.

**Comment by samkit-jain** [comment, marked as answer]
Hi @LeeSugden Appreciate your interest in the library. I tried with a different approach and it works fine. Here it is

```python
from tqdm import tqdm
import pdfplumber

def progress_bar_test():
	pdf = pdfplumber.open("file.pdf")
	pbar = tqdm(total=len(pdf.pages))

	for page in pdf.pages:
		text = page.extract_text()
		pbar.update(1)

	pbar.close()

if __name__ == "__main__":
	progress_bar_test()
```

The final output was
```bash
$ python main.py 
100%|█████████████████████████████████████████████| 8/8 [00:02<00:00,  3.49it/s]
```

It goes from 1 to the number of pages but you can easily configure it if you want to go from 10 to 100.
